cmd/tempo/app/server_service.go

```diff
@@ -71,22 +77,28 @@ func (s *tempoServer) StartAndReturnService(cfg server.Config, supportGRPCOnHTTP
 	var err error
 
 	metrics := server.NewServerMetrics(cfg)
-	// use tempo's mux unless we are doing grpc over http, then we will let the library instantiate its own
-	// router and piggy back on it to route grpc requests
-	cfg.Router = s.mux
-	if supportGRPCOnHTTP {
+	DisableSignalHandling(&cfg)
+
+	if !supportGRPCOnHTTP {
+		// We don't do any GRPC handling, let the library handle all routing for us
+		cfg.Router = s.mux
+
+		s.externalServer, err = server.NewWithMetrics(cfg, metrics)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create server: %w", err)
+		}
+		s.handler = s.externalServer.HTTPServer.Handler
```

Comment:
@joe-elliott we were really close while pairing. We used `s.externalServer.HTTP` which is the router, `HTTPServer.Handler` is the handler which contains all middleware as well.

Reply:
so close!